sklearn/linear_model/_coordinate_descent.py

```diff
@@ -974,30 +974,29 @@ def fit(self, X, y, sample_weight=None, check_input=True):
             # Long: The objective function of Enet
             #
             #    1/2 * np.average(squared error, weights=sw)
-            #    + alpha * penalty                                   (1)
+            #    + alpha * penalty                                             (1)
             #
             # is invariant under rescaling of sw.
             # But enet_path coordinate descent minimizes
             #
-            #     1/2 * sum(squared error) + alpha * penalty
+            #     1/2 * sum(squared error) + alpha' * penalty                  (2)
             #
             # and therefore sets
             #
-            #     alpha = n_samples * alpha
+            #     alpha' = n_samples * alpha                                   (3)
             #
-            # inside its function body, which results in an objective
-            # equivalent to (1) without sw.
+            # inside its function body, which results in objective (2) being
+            # equivalent to (1) in case of no sw.
             # With sw, however, enet_path should set
             #
-            #     alpha = sum(sw) * alpha                            (2)
+            #     alpha' = sum(sw) * alpha                                     (4)
             #
-            # Therefore, using the freedom of Eq. (1) to rescale alpha before
-            # calling enet_path, we do
+            # Therefore, we use the freedom of Eq. (1) to rescale sw before
+            # calling enet_path, i.e.
             #
-            #     alpha = sum(sw) / n_samples * alpha
+            #     sw = n_samples / sum(sw) * sw
```

Comment:
I think this is slightly clearer?

```suggestion
            #     sw *= n_samples / sum(sw)
```

and consistent with the `alpha *=` a few lines below.